packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -289,8 +289,14 @@ module A = {
       ))
       |> Rationale.Result.return
     }
-  let rangeFloat = (~step=1, start, stop) =>
-    Belt.Array.rangeBy(start, stop, ~step) |> fmap(Belt.Int.toFloat)
+
+  let diff = (arr: array<float>): array<float> =>
+    Belt.Array.zipBy(arr, Belt.Array.sliceToEnd(arr, 1), (left, right) => right -. left)
+
+  let rec rangeFloat = (~step: float=1.0, start : float, end: float) : array<float> => 
```

Comment:
I suggest giving an example for how this works, in the comments. I've gotten bitten by these before.

Like, 
//rangeFloat(~step=0.5, start: 0, end: 2) => [0, 0.5, 1.0, 1.5, 2.0]

Reply:
Another name for this function, given that it's in Array, would be ``fromRange``. Also, in that case, it could go in Array.Float

Reply:
I removed this one as well